### PR TITLE
Hadoop improvements

### PR DESCRIPTION
**Pull request by wendigo**
If it's ok I'll do the same for other environments

### CODE REVIEW

File: prestodev/cdh5.12-hive/files/etc/hadoop/conf/yarn-site.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
     <property>
         <name>yarn.log-aggregation-enable</name>
-        <value>true</value>
+        <value>false</value>
```
Comment (losipiuk) [inline prestodev/cdh5.12-hive/files/etc/hadoop/conf/yarn-site.xml:32]: where do container logs land if aggregation is disabled. Is the path already covered by list to download logs from you added?

File: prestodev/cdh5.15-hive/files/etc/hadoop-health.d/supervisord.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+fi
+
+FAILED=$(supervisorctl status | grep -v RUNNING || true)
+
+if [ "$FAILED" == "" ]; then
+  exit 0
+else
+  echo "Some of the services are failing: ${FAILED}"
+  exit 1
+fi
```
Comment (losipiuk) [inline prestodev/cdh5.15-hive/files/etc/hadoop-health.d/supervisord.sh:15]: new line

File: prestodev/cdh5.15-hive/files/etc/hadoop-health.d/supervisord.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+set -euo pipefail
```
Comment (losipiuk) [inline prestodev/cdh5.15-hive/files/etc/hadoop-health.d/supervisord.sh:1]: #!/usr/bin/env bash

File: prestodev/cdh5.15-hive/files/etc/hadoop-init.d/empty.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+echo "Running hadoop init.d"
```
Comment (losipiuk) [inline prestodev/cdh5.15-hive/files/etc/hadoop-init.d/empty.sh:3]: new liune

File: prestodev/cdh5.15-hive/files/usr/bin/health.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+fi
+
+set -x
+
+if test -d /etc/hadoop-health.d; then
+    for health_script in /etc/hadoop-health.d/*; do
+        echo "Running Hadoop health check ${health_script}" &>> /var/log/container-health.log
+        "${health_script}" &>> /var/log/container-health.log || exit 1
+    done
+fi
```
Comment (losipiuk) [inline prestodev/cdh5.15-hive/files/usr/bin/health.sh:16]: new line

File: prestodev/cdh5.15-hive/files/usr/bin/run-hadoop.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        echo "Running Hadoop init script ${init_script}"
+        "${init_script}"
+    done
+fi
+
+trap exit INT
+
+echo "Running services with supervisord"
+
+supervisord -c /etc/supervisord.conf
```
Comment (losipiuk) [inline prestodev/cdh5.15-hive/files/usr/bin/run-hadoop.sh:22]: new line

**Review by losipiuk — APPROVED** [review]

File: prestodev/cdh5.12-hive/files/etc/hadoop/conf/yarn-site.xml (hunk 1 above)
Comment (wendigo) [reply prestodev/cdh5.12-hive/files/etc/hadoop/conf/yarn-site.xml:32]:
```
<property>
        <description>Where to store container logs.</description>
        <name>yarn.nodemanager.log-dirs</name>
        <value>/var/log/hadoop-yarn/containers</value>
    </property>

    <property>
        <description>Where to aggregate logs to.</description>
        <name>yarn.nodemanager.remote-app-log-dir</name>
        <value>/var/log/hadoop-yarn/apps</value>
    </property>
```

Yeah. It's already covered

File: prestodev/gpdb-6/Dockerfile [hunk 7]
```diff
@@ -4,8 +4,8 @@ SHELL ["/bin/bash", "-c"]
 
 # Version arguments and install dir, GPHOME is symlink
 ARG GPDB_MAJOR_VERSION=6
-ARG GPDB_VERSION=6.11.0-1
-ARG GPDB_DIR=/opt/greenplum-db-$GPDB_MAJOR_VERSION-6.11.0
+ARG GPDB_VERSION=6.11.1-1
```
Comment (losipiuk) [inline prestodev/gpdb-6/Dockerfile:7]: again ? :)

Comment (wendigo) [reply prestodev/gpdb-6/Dockerfile:7]: cc: @adamjshook
@losipiuk yeah, again :)

File: prestodev/hdp2.6-hive/files/etc/hadoop-init.d/01-empty.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
```
Comment (losipiuk) [inline prestodev/hdp2.6-hive/files/etc/hadoop-init.d/01-empty.sh:1]: Why do we need this one?
I think greeting should be printed by `run-hadoop.sh`

File: prestodev/gpdb-6/Dockerfile (hunk 7 above)
Comment (adamjshook) [reply prestodev/gpdb-6/Dockerfile:7]: Oh boy. Fixing this today.

**Pull request closed without merging**